src/main/java/com/google/devtools/build/lib/remote/blobstore/http/DownloadCommand.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.remote.blobstore.http;
+
+import java.io.OutputStream;
+import java.net.URI;
+
+class DownloadCommand {
+
+  private final URI uri;
+  private final boolean casDownload;
+  private final String hash;
+  private final OutputStream out;
+
+  protected DownloadCommand(URI uri, boolean casDownload, String hash, OutputStream out) {
+    this.uri = uri;
```

Comment:
Which of these may be null, which don't? Add Preconditions.checkNotNull for those that don't, maybe use NullPointerTester to check for that.

Reply:
none of them. done.